### PR TITLE
disper: repair missing libXrandr and libX11 dependencies

### PR DESCRIPTION
**Pull request by grahamc**
###### Things done
- [X] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [X] NixOS
  - [ ] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

---

This change is [Reviewable](https://reviewable.io/reviews/nixos/nixpkgs/16438)

### CODE REVIEW

File: pkgs/tools/misc/disper/default.nix [hunk 1]
```diff
@@ -1,14 +1,20 @@
...
 
 stdenv.mkDerivation rec {
   name = "disper-0.3.1";
 
-  buildInputs = [python];
+  buildInputs = [python makeWrapper];
 
   preConfigure = ''
     export makeFlags="PREFIX=$out"
   '';
```
Comment (vrthra) [inline pkgs/tools/misc/disper/default.nix]: you could use the `makeFlags` in Nix directly.

**Comment by fpletz** [comment]
With or without this PR, `disper -l` fails with the following error for me:

    No suitable backend found

**Comment by rycee** [comment]
Seems to be an issue with multiple outputs, `disper -l` works for me with the following change to the PR:

```diff
diff --git a/pkgs/tools/misc/disper/default.nix b/pkgs/tools/misc/disper/default.nix
index ea503b9..5042c33 100644
--- a/pkgs/tools/misc/disper/default.nix
+++ b/pkgs/tools/misc/disper/default.nix
@@ -11,7 +11,7 @@ stdenv.mkDerivation rec {
 
   postInstall = ''
       wrapProgram $out/bin/disper \
-      --prefix "LD_LIBRARY_PATH" : "${xorg.libXrandr}/lib:${xorg.libX11.out}/lib"
+      --prefix "LD_LIBRARY_PATH" : "${xorg.libXrandr.out}/lib:${xorg.libX11.out}/lib"
 
   '';
```

**Comment by grahamc** [comment]
Nice catch, @rycee. I must have missed than when I ported it from 16.03 to unstable. Thanks!

Updated the PR to include the fix. @fpletz, can you try again?

---

Review status: 0 of 1 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/nixos/nixpkgs/16438#-:-KMlgnbvrnAx1nW6zS_y:1726864707)*

**Comment by rycee** [comment]
I've merged this into master in commit 32744b72e5a048fea7250c444e73d2c0b605d50b. Thanks for the contribution!

**Comment by grahamc** [comment]
Thank you!

**Pull request closed without merging**